src/21-pubsub/index.md

```diff
@@ -0,0 +1,134 @@
+
+```sh
+> ipfs daemon --enable-pubsub-experiment
+```
+
+This will tell ipfs to create and enable the pubsub service. It also implies
+that you will only be able to use pubsub with other peers who choose to enable
+it.
+
+To subscribe to the topic `foo`, run:
+```sh
+> ipfs pubsub sub foo
+```
+
+Now, any messages for the topic `foo` will print to your console.
+
+To publish a message to the topic `foo`, open up another terminal and run:
+```sh
+> ipfs pubsub pub foo "hello world"
+```
+
+You should see "hello world" printed out in the first terminal.  You can also
+run the `pub` command on any other connected ipfs node and your node will
+receive the message.  Messages are routed through connected, subscribed peers.
+This means that if peers A, B, and C are all subscribed to `foo`, A is connected
+to B, and B is connected to C, but A is not directly connected to C, A will
+still receive messages that C published to `foo` through B. This can be very
+useful for routing messages in networks with poor NAT traversal or otherwise
+suboptimal connectivity.
+
+To see all peers with pubsub enabled, check the output of:
+```sh
+> ipfs pubsub peers
+```
+
+To see all the topics you are currently subscribed to, run:
+```sh
+> ipfs pubsub ls
+```
+
+## Pubsub in the wild
+As an example, we have integrated pubsub into [Orbit](https://github.com/orbitdb/orbit).
+This allows Orbit to provide a fully distributed, peer-to-peer chat without *any*
+server anywhere. We are also actively working to put Conflict-Free Replicated Data
+Types (CRDTs) on IPFS pubsub using libraries like Y.js and swarm.js.
+
+Together, pubsub and CRDTs open new doors for collaborative editing of distributed
+content. We are working with [@edsilv](https://github.com/edsilv) and
+[@aeschylus](https://github.com/aeschylus) to prepare a demo of IPFS in two
+[IIIF](http://iiif.io/about/) image viewers
+[#240](https://github.com/ipfs/notes/issues/240). This will showcase how to
+collaboratively annotate images from repositories dispersed around the world.
+The demo will take place at the [IIIF 2017 Conference](https://2017iiifconferencethevatican.sched.com/event/AChW/presentation-interoperable-peer-to-peer-research-with-iiif-and-ipfs-room-5.0) in The Vatican.
+We will publish a video of the demo, along with all of the code.
+
+## What's next?
+The next two areas of focus for ipfs pubsub are authentication and message routing.
+
+Currently, any peer can publish to any pubsub topic. We plan to implement an
+authenticated mode for pubsub topics, where only authorized peers — those given a
+cryptographic key or capability — can publish messages. We are still working out
+the sharing and capability granting model.
+
+After that, we plan to improve the message routing algorithms.
+The current routing algorithm floods messages to every subscriber, resulting in
+some peers receiving the same message multiple times. Finding a better routing
```

Comment:
~Finding a better routing algorithm would go a long way towards reducing that overhead.~
We affectionately call this approach "floodsub". We plan to replace it with a more efficient routing algorithm, which will go a long way towards reducing overhead and improving scalability.